src/Components/Server/src/Circuits/RemoteRenderer.cs

```diff
@@ -340,15 +340,16 @@ public UnacknowledgedRenderBatch(long batchId, ArrayBuilder<byte> data, TaskComp
             public ValueStopwatch ValueStopwatch { get; }
         }
 
-        private void CaptureAsyncExceptions(Task task)
+        private async void CaptureAsyncExceptions(ValueTask task)
         {
-            task.ContinueWith(t =>
+            try
             {
-                if (t.IsFaulted)
-                {
-                    UnhandledException?.Invoke(this, t.Exception);
-                }
-            });
+                await task;
+            }
+            catch (Exception ex)
+            {
+                UnhandledException?.Invoke(this, ex);
+            }
```

Comment:
There's a subtle difference in behaviour hiding here. If `UnhandledException` throws this will now crash the app.

Reply:
This also now behaves differently for cancelled tasks.